src/ProjectService.Business/Commands/Task/EditTaskCommand.cs

```diff
@@ -76,44 +76,32 @@ private IGetDepartmentResponse GetDepartment(Guid userId, List<string> errors)
             _userRepository = userRepository;
             _validator = validator;
             _accessValidator = accessValidator;
-            _httpContext = httpContextAccessor.HttpContext;
+            _httpContextAccessor = httpContextAccessor;
             _mapper = mapper;
             _logger = logger;
             _requestClient = requestClient;
         }
 
         public OperationResultResponse<bool> Execute(Guid taskId, JsonPatchDocument<EditTaskRequest> patch)
         {
-            _validator.ValidateAndThrowCustom(patch);
-
             var errors = new List<string>();
 
             DbTask task = _taskRepository.Get(taskId, false);
-            List<DbProjectUser> projectUsers =
-                _userRepository.GetProjectUsers(task.ProjectId, false).ToList();
-
-            Guid requestUserId = _httpContext.GetUserId();
-            IGetDepartmentResponse department = GetDepartment(requestUserId, errors);
-
-            bool isAdmin = _accessValidator.IsAdmin();
 
-            bool isProjectParticipant = projectUsers.FirstOrDefault(x =>
-                x.UserId == requestUserId) != null;
+            Guid requestUserId = _httpContextAccessor.HttpContext.GetUserId();
 
-            bool isDepartmentDirector = false;
-            if (department != null)
-            {
-                 isDepartmentDirector = department.DirectorUserId == requestUserId;
-            }
-
-            if (!isAdmin && !isProjectParticipant && !isDepartmentDirector)
+            if (!_accessValidator.IsAdmin()
+                && !(_userRepository
```

Comment:
we have to rework this check, need to create a method in the repository, which will try to find one record of project employee. Get all list of project users here is bad, I guess.